### PR TITLE
Use `sha512` for hashing (fixes support for Node 18)

### PR DESCRIPTION
**Pull request by MelSumner**
If merged, this PR: 
- Updates package from using md4 (this was causing node 18 issues) 
- Updates some dependencies identified as "should be updated"
- Removes yarn since npm was being used first
- Removes appveyor

Tests pass.

### CODE REVIEW

**Comment by MelSumner** [comment]
@rwjblue should I also change from travis to github actions? WDYT?

**Comment by rwjblue** [comment]
Yeah, we will need to migrate to GH actions for sure. It's probably better to do that before we land this change (just so that we actually have some CI coverage...).

**Comment by rwjblue** [comment]
Landed GH actions CI in https://github.com/broccolijs/broccoli-multifilter/pull/9

File: package.json [hunk 1]
```diff
@@ -26,11 +26,11 @@
     "promise-map-series": "^0.2.1"
   },
   "devDependencies": {
-    "broccoli-fixture": "^1.0.0",
+    "broccoli-fixture": "^0.1.0",
```
Comment (rwjblue) [inline package.json]: This seems strange, why would we roll back to 0.1.0 here?

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (MelSumner) [reply package.json]: no idea, reverted it though.

File: yarn.lock
Comment (rwjblue) [inline yarn.lock:1]: I think we still use `yarn` in CI, so I think this file needs to come back

File: package-lock.json
Comment (rwjblue) [inline package-lock.json:1]: CI is using `yarn` still, so we should probably delete this

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: yarn.lock
Comment (MelSumner) [reply yarn.lock:1]: ok swapped npm for yarn

File: package-lock.json
Comment (MelSumner) [reply package-lock.json:1]: ok 👍

**Pull request merged**